pkg/synchronization/controller.go

```diff
@@ -534,6 +539,38 @@ func (c *Controller) createOrUpdateSecret(logger *zap.Logger, secretSpec *core_v
 	return true, secret, err
 }
 
+func (c *Controller) ensureCommonSecretExists(logger *zap.Logger, ns *core_v1.Namespace) (bool /* retriable */, error) {
+	_, err := c.MainClient.CoreV1().Secrets(ns.Name).Get(apik8scompute.CommonSecretName, meta_v1.GetOptions{})
+
+	// nothing to do, it exists; nothing to update
+	if err == nil {
+		logger.Sugar().Infof("Secret named %q already exists", apik8scompute.CommonSecretName)
```

Comment:
To think about in future: the secret may be outdated?

Reply:
In general since it has no contents at creation time, it can't really be outdated. Or is the worry that it somehow hangs around if they delete/recreate everything? If the namespace gets deleted, the secret should be deleted, too.

Reply:
sorry, I missed that we create an empty secret, all good.

Reply:
This should not log at all, or do so on debug to avoid spamming too much.

Reply:
Updated to Debug.